synapse/storage/databases/main/event_federation.py

```diff
@@ -60,6 +70,14 @@
 logger = logging.getLogger(__name__)
 
 
+# All the info we need while iterating the DAG while backfilling
+class BackfillQueueNavigationItem(NamedTuple):
+    depth: int
+    stream_ordering: int
+    event_id: str
+    type: str
```

Comment:
Instead on `NamedTuple` we tend to use `attrs`, which gives us a bit more control:

```suggestion
@attr.s(frozen=True, slots=True, auto_attribs=True)
class BackfillQueueNavigationItem:
    depth: int
    stream_ordering: int
    event_id: str
    type: str
```